openpnp/openpnp
# Improvements for BottomVision Calibration (issue #235)

# Description
- Improvement/Fix for BottomVision Calibration (issue #235)
   - Enable BottomVision Calibration
   - Add parameteres (machine.xml) for hardcoded numbers (eg. rotation limits)
   - new  Parameters (machine.xml):
      - angleAverage ... 1 ... means no averaging.
      - angleStart ... 0.0
      - angleStop ... 360.0
      - angleIncrement ... 15.0
   - All new parameters are initialized by the previous hardcoded values. 
   - Show 0degree  measurement in new fields next to the calibrate button
   - add findCircleAveraged(int avg) around findCircle() in case of difficult light conditions
- Implement issue #784 

# Justification
At least in my setup it seems to be working and brings my DebugNozzle-tip (1.5mm runout) back to an error of 0.1mm after changing the nozzle-tip with the tool-changer.

# Instructions for Use
Shall work as designed before pluse some parameters in machine.xml.

# Implementation Details
0. Detailed change list is the commit comment 6ed5881 ... sorted by line numbers
1. How did you test the change? Be descriptive. Untested code will not be accepted.
   - MyTestProcedure-1:
      - Set Reticle/Fiducial and Reticle/Options/size=nozzleTip outer diameter
      - Run the Calibration
      - Go to BottomCamera and jog C for (angleStart->angleStop by angleIncrement) ... nozzle-tip should be within/close to the Fiducial
2. Did you follow the [coding style](https://github.com/openpnp/openpnp/wiki/Developers-Guide#coding-style)?
Tried my best. Be aware of a space/tab respective line-ending issue affecting the entire file.
3.  `org.openpnp.spi` or `org.openpnp.model` packages no touched
4. Be sure to run `mvn test` before submitting the Pull Request. If the tests do not pass the Pull Request will not be accepted.
   - run 13/13 ... 0 errors ... 0 failures
